kodi_addon_checker/check_addon.py

```diff
@@ -384,3 +391,33 @@ def _check_dependencies(report: Report, addon_path, repo_addons):
             if LooseVersion(available_version) < LooseVersion(required_version) and (required_addon not in ignore):
                 report.add(Record(PROBLEM, "Version mismatch for addon %s. Required: %s, Available: %s "
                                   % (required_addon, required_version, available_version)))
+
+
+def get_addon_name(xml_path):
+    tree = ET.parse(xml_path).getroot()
+    return (tree.get("id"), tree.get("version"))
+
+
+def _check_for_existing_addon(report: Report, addon_path, addon_branch):
+    """Check if addon submitted already exists or not"""
+    addon_xml = os.path.join(addon_path, "addon.xml")
+    addon_name, addon_version = get_addon_name(addon_xml)
+
+    branches = ['gotham', 'helix', 'isengard', 'jarvis', 'krypton', 'leia']
+
+    for branch in branches:
+
+        branch_url = ROOT_URL.format(branch=branch)
+        repo_addons = _get_addons(branch_url)
+
+        if addon_name in repo_addons:
+            if LooseVersion(addon_version) == LooseVersion(repo_addons[addon_name]):
+                report.add(Record(PROBLEM, "%s addon already exist with current version in lower branch"
+                                  % addon_name))
+
+            else:
+                report.add(Record(INFORMATION, "%s addon also exists in %s branch but with lower version"
```

Comment:
as you compare for equality in the if statement the addon repo version could be higher too

Reply:
oh instead of having `==` I should have `<=` so if the repo_version will be higher or equal then it will give `addon already exist with current version in lower branch` right ?